invoke-rune.lic

```diff
@@ -0,0 +1,134 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#focus-rune
+=end
+
+custom_require.call(%w[common-travel common common-items common-arcana spellmonitor])
+
+class Runestone
+  include DRC
+  include DRCT
+  include DRCI
+  include DRCA
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'spell',
+          regex: /^[A-z\s\-']+$/i,
+          variable: true,
+          description: "Specify spell to cast from runestone. Spells are: 'Arc Light'
+                Althleticism Bless Calm 'Clear Vision' Compost 'Eagle's Cry'
+                'Fire Shards' 'Glythtide's Gift' Geyser 'Protection from Evil'
+                Refresh Shadows Zephyr. Use double quotes." },
+
+        { name: 'debug',
+          regex: /debug/i,
+          optional: true }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+    spell = args.spell
+    @debug = args.debug
+    settings = get_settings
+    buff_spell_mana_threshold = settings.buff_spell_mana_threshold
+    @runestone_storage = settings.runestone_storage
+    @harness = settings.runestone_harness
+    runestone_purchase = settings.runestone_purchase
+
+    echo spell if @debug
+    echo @harness if debug
+    @rune = get_rune_name_for_spell_name(spell)
+    echo @rune if @debug
+    spell = format_spell_name(args.spell)
+
+    until DRSpells.active_spells.include?(spell)
+      pause 1 while mana < buff_spell_mana_threshold
+      if DRCI.exists?("#{@rune} runestone in #{@runestone_storage}")
+        runestone_cast
+      elsif runestone_purchase
+        buy_runestone
+      else
+        out_of_runestone
+      end
+    end
+  end
+
+  def format_spell_name(spell)
```

Comment:
Why not just downcase all spell names instead?

Reply:
This checks against the game's name for it. I could play around with downcase this.

Reply:
Downcase was not working for me. I kept getting undefined variable downcase.